spec/request/request_ip_spec.rb

```diff
@@ -0,0 +1,163 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.shared_examples "expected request IP address" do 
+  it "uses the expected request IP address" do
+    get '/request_ip', env: request_env, headers: headers
+    expect(body_hash['ip']).to eq(expected_request_ip)
+  end
+end
+
+RSpec.describe "request IP address determination", :type => :request do
+  # We inject this RequestIpController to provide our tests with information
+  # about the IP address available to Rails controllers
+  class RequestIpController < ApplicationController
+    def echo
+      render json: { ip: request.ip }
+    end
+  end
+
+  before do
+    # Add the test endpoint to the routes
+    Rails.application.routes.draw do
+      get '/request_ip' => 'request_ip#echo'
+    end
+
+    # Configure Slosilo to produce valid access tokens
+    Slosilo["authn:rspec"] ||= Slosilo::Key.new
+
+    # Set the TRUSTED_PROXIES environment variable for the test
+    ENV['TRUSTED_PROXIES'] = trusted_proxies if trusted_proxies
+  end
+
+  after do
+    # Reset the Conjur routes to remove the test endpoint
+    Rails.application.reload_routes!
+  end
+
+  # --------------------------------------------------------------------
+  # Test Setup
+  # --------------------------------------------------------------------
+
+  let(:body_hash) { JSON.parse(response.body) }
+
+  # We can't modify the access token middleware to add an exception to our
+  # test route, so we need to create an access token for our request to use.
+  let(:bearer_token) { Slosilo["authn:rspec"].signed_token('admin') }
+  let(:token_auth_header) do
+    "Token token=\"#{Base64.strict_encode64 bearer_token.to_json}\""
+  end
+
+  let(:request_env) do
+    { 'HTTP_AUTHORIZATION' => token_auth_header, 'REMOTE_ADDR' => remote_ip }
+  end
+
+  let(:headers) do
+    {}.tap do |h|
+      h['X-Forwarded-For'] = x_forwarded_for if x_forwarded_for
+    end
+  end
+
+  # The value to user for the TRUSTED_PROXIES environment variable
+  let(:trusted_proxies) { nil }
+
+  # The value to use for the X-Forwarded-For HTTP header
+  let(:x_forwarded_for) { nil }
+
+  # --------------------------------------------------------------------
+  # Test Scenarios
```

Comment:
I'm missing a test with CIDR range in the trusted proxies

Reply:
Thanks, I've added tests to include trusted IP range examples using CIDR notation.